script.js

```diff
@@ -1,4 +1,137 @@
+const invalidCases = (a) =>
+  a.toLowerCase() !== "paper" &&
+  a.toLowerCase() !== "rock" &&
+  a.toLowerCase() !== "scissors";
+// All cases where user wins
+const winCases = (b, c) =>
+  (b.toLowerCase() === "paper" && c === "rock") ||
+  (b.toLowerCase() === "rock" && c === "scissors") ||
+  (b.toLowerCase() === "scissors" && c === "paper");
+// All cases where both parties choose the same object.
+const drawCases = (d, e) => d.toLowerCase() === e;
+
+// normal reverse korean mode game play
+const normalReversedKoreanGamePlay = (userInput, isReversed, isKorean) => {
+  let showMeHand = getRandomHand();
+  let winner;
+  //If invalid input , return feedback.
+  if (invalidCases(userInput)) {
+    return invalidMessage(userInput);
+  }
+  //If both parties choose the same object, it's a draw.
+  else if (drawCases(userInput, showMeHand)) {
+    winner = winnerOfGame();
+    //return ultimate winner declaration when it is korean mode
+    return isKorean
+      ? koreanDrawMessage(userInput, showMeHand, userWin, systemWin, winner)
+      : drawMessage(userInput, showMeHand, userWin, systemWin, winner);
+  }
+  // User wins.
+  else if (isReversed ^ winCases(userInput, showMeHand)) {
+    //increment userWin score
+    userWin += 1;
+    winner = winnerOfGame();
+    return isKorean
+      ? koreanWinMessage(userInput, showMeHand, userWin, systemWin, winner)
+      : winMessage(userInput, showMeHand, userWin, systemWin, winner);
+  }
+  // System wins.
+  else if (!(isReversed ^ winCases(userInput, showMeHand))) {
+    //increment systemWin score
+    systemWin += 1;
+    winner = winnerOfGame();
+    return isKorean
+      ? koreanLoseMessage(userInput, showMeHand, userWin, systemWin, winner)
+      : loseMessage(userInput, showMeHand, userWin, systemWin, winner);
+  }
+};
+
+//track scores
+let userWin = 0;
+let systemWin = 0;
+//track winner
+const winnerOfGame = () =>
+  systemWin > userWin ? "System" : systemWin < userWin ? "You" : "Nobody";
+//reset Game when switch mode
+const resetGame = () => {
+  userWin = 0;
+  systemWin = 0;
+};
+//define default game mode as pending for username
+let userName = "";
+let gameMode = "pending for username";
+
+// Main game
+const main = (input) => {
+  // waiting for user to enter username
+  if (gameMode == "pending for username") {
+    userName = input;
+    gameMode = "normal";
+    return `Welcome ${userName}.\<br>\You may choose to play normal, reverse, or korean game.`;
+  }
+  //user input to switch game mode
+  if (
+    input.toLowerCase() == "reverse" ||
+    input.toLowerCase() == "normal" ||
+    input.toLowerCase() == "korean"
+  ) {
+    gameMode = input.toLowerCase();
+    resetGame();
+    return `Welcome to ${input.toLowerCase()} mode`;
+  }
+  // game mode to execute
+  switch (gameMode) {
+    case "normal":
+      return normalReversedKoreanGamePlay(input, false, false);
+    case "reverse":
+      return normalReversedKoreanGamePlay(input, true, false);
```

Comment:
Great job declaring in your initial `normalReversedKoreanGamePlay` function,
I think the boolean variables within the function is clear, what would help further especially when you refer back to this is to have comments just above these function calls

Could you think of a way to have`normalReversedKoreanGamePlay` be shortened in terms of number of characters used?